c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -388,25 +388,36 @@ class THCCachingAllocator {
     return basePtr;
   }
 
-  void recordStream(void* ptr, cuda::CUDAStream stream) {
+  void recordStream(const DataPtr& ptr, cuda::CUDAStream stream) {
     // Empty tensor's storage().data() might be a null ptr. As there is no
     // blocks associated with those tensors, it is fine to do nothing here.
-    if (!ptr) {
+    if (!ptr.get()) {
       return;
     }
 
+    // The ptr is not allocated by CUDA, most likely user error
+    TORCH_CHECK(
+        ptr.device().type() == kCUDA, "Tensor is not allocated in CUDA");
```

Comment:
You're planning to delete this now, is that right?

Reply:
Actually, I am still looking into it, when the HIP version replaces the kCUDA enum with kHIP, which I think should be the case in ROCm/HIP machines. 
I have found an AMD machine and am trying to build and test the HIP version over there, I am still having trouble building Pytorch on that particular machine, I will continue the investigation this Monday.

Reply:
@ezyang if you know why the check won't pass on a rocm machine, please let me know

Reply:
I don't remember if HIP tensors report as CUDA or HIP device type in the AMD world. If they report as HIP device type you have no problems and this line will work. But since this is a bootcamp task, my preference is to reduce the amount of work that you need to do, and you can reduce the amount of work by just not doing extra stuff that isn't required by the task.

Reply:
@ezyang I think you are right, even when we run the code on AMD machines, the device type for tensor is still kCUDA, https://github.com/pytorch/pytorch/blob/master/caffe2/contrib/aten/aten_op_template.h#L61
I will simply remove the line.

Reply:
Curious, when are we going to use `kHIP`?